intelmq_webinput_csv/app.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017-2018 nic.at GmbH <wagner@cert.at>
+# SPDX-License-Identifier: AGPL-3.0
+import json
+import pkg_resources
+import traceback
+import logging
+import os
+
+from flask import Flask, make_response, request
+
+from intelmq import HARMONIZATION_CONF_FILE
+from intelmq.lib.harmonization import DateTime, IPAddress
+from intelmq.bots.experts.taxonomy.expert import TAXONOMY
+from intelmq.lib.message import MessageFactory
+from intelmq.lib.pipeline import PipelineFactory
+from intelmq.lib.utils import load_configuration
+
+from intelmq_webinput_csv.version import __version__
+from intelmq_webinput_csv.lib import util
+from intelmq_webinput_csv.lib.exceptions import InvalidCellException
+from intelmq_webinput_csv.lib.csv import CSV
+
+CONFIG_FILE = os.path.join('/config/configs/webinput', 'webinput_csv.conf')
+logging.info('Reading configuration from %r.', CONFIG_FILE)
+with open(CONFIG_FILE) as handle:
+    CONFIG = json.load(handle)
+    BASE_URL = CONFIG.get('base_url', '')
+    if BASE_URL.endswith('/'):
+        BASE_URL = BASE_URL[:-1]
+
+CUSTOM_FIELDS_HTML_TEMPLATE = """
+<div class="field">
+    <div class="control">
+        <label class="label">{name}</label>
+        <input class="input" type="text" placeholder="{name}" v-model="previewFormData.{jsname}">
+    </div>
+</div>"""
+CUSTOM_FIELDS_JS_DEFAULT_TEMPLATE = "{jsname}: '{default}',"
+CUSTOM_FIELDS_JS_FORM_TEMPLATE = "formData.append('custom_{name}', this.previewFormData.{jsname});"
+custom_fields_html = []
+custom_fields_js_default = []
+custom_fields_js_form = []
+for key, value in CONFIG.get('custom_input_fields', {}).items():
+    jskey = 'custom' + key.title().replace('.', '')
+    custom_fields_html.append(CUSTOM_FIELDS_HTML_TEMPLATE.format(name=key, jsname=jskey))
+    custom_fields_js_default.append(CUSTOM_FIELDS_JS_DEFAULT_TEMPLATE.format(jsname=jskey, default=value))
+    custom_fields_js_form.append(CUSTOM_FIELDS_JS_FORM_TEMPLATE.format(name=key, jsname=jskey))
+
+STATIC_FILES = {
+    'js/preview.js': None,
+    'js/upload.js': None,
+    'preview.html': None,
+    'index.html': None,
+}
+
+for static_file in STATIC_FILES.keys():
+    filename = pkg_resources.resource_filename('intelmq_webinput_csv', 'static/%s' % static_file)
```

Comment:
f-string would be more readable

Reply:
This is still leftover from the initial code base but will fix.